### ISSUE TITLE
Java DSL for web services

### ISSUE THREAD

**Issue by afattahi54** [spring-projects/spring-integration#2818]
We are starting a new project with spring Integration.
We found that it is recommended to use Java DSL to configure channels.Well, there are lots of first class integration for `Amqp`, `Feed`, `Jms`, `Files`, `(S)Ftp`, `Http`, `JPA`, `MongoDb`, `TCP/UDP`, `Mail`.
We need to integrate with `web services` (  `inboud` and `out bound`)

Is there any planning for implementing WebService DSL. 

Thanks !

**Comment by artembilan** [comment]
We have those plans, indeed: https://jira.spring.io/browse/INT-4429.

Only the problem that it is not so important in our backlog.

We are always open for contributions, though: https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc

You still can use existing Web Service components in Java DSL with regular `@Bean` definitions: https://docs.spring.io/spring-integration/docs/current/reference/html/#java-dsl-protocol-adapters

**Comment by afattahi54** [comment]
Thanks ... we are looking forward that

**Comment by sirimamilla** [comment]
Hi @artembilan ,
We are also having some projects in need of dsl ws. Can this be prioritised?

**Comment by amalajone** [comment]
Hi @artembilan 
Even in our project ,we are looking forward to have Java DSL for WS  .Can you please prioritise this ?

**Comment by artembilan** [comment]
Well, while I’m sleeping, please, consider my suggestion above regarding contribution. I mean if you strongly need it, please, consider to provide a solution. It is not so hard to follow similar implementations for HTTP or WebFlux. Otherwise I’ll come back to this somewhere next week.

Thanks for understanding!

**Comment by garyrussell** [comment]
I have some time to work on this.
